src/lsp/cobol_ptree/terms.ml

```diff
@@ -60,15 +60,27 @@ type alphanum_quote =
   | Double_quote (* "..." *)
 [@@deriving ord]
 
+type alphanum_repr =
+  | Native_bytes
+  | Null_terminated_bytes
+[@@deriving ord]
+
 type alphanum_string =
   {
     str: string;
     quotation: alphanum_quote;
     hexadecimal: bool;
+    runtime_repr: alphanum_repr;
   }
 [@@deriving ord]
 
-let pp_alphanum_string ppf { hexadecimal; quotation; str } =
+let pp_alphanum_repr ppf repr =
```

Comment:
Not sure doing nothing in some/most cases is what we'd want: I think that could become confusing. I'd go with local decisions on the string representation: one inside `pp_alphanum_string` below, and the other one (even if that's similar) at the other place where `pp_alphanum_repr` is used.